### ISSUE TITLE
Compute V2: wrong secgroup, when new compute instance is attached to a port

### ISSUE THREAD

**Issue by kayrus** [terraform-providers/terraform-provider-openstack#754]
### Affected Resource(s)

- openstack_compute_instance_v2

### Terraform Configuration Files
```hcl
resource "openstack_networking_port_v2" "port_1" {
  network_id = "${openstack_networking_network_v2.network_1.id}"

  fixed_ip = {
    subnet_id = "${openstack_networking_subnet_v2.subnet_1.id}"
  }
}

resource "openstack_compute_instance_v2" "test" {
  name = "test"

  security_groups = ["test"]

  flavor_name = "m1.small"
  image_name  = "ubuntu-16.04-amd64"

  network {
    port = "${openstack_networking_port_v2.port_1.id}"
  }
}
```

### Expected Behavior

second run should not offer to modify the security group

### Actual Behavior

second run offers to modify the instance security group

```
An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ openstack_compute_instance_v2.test
      security_groups.3632233996: "" => "test"
      security_groups.3814588639: "default" => ""


Plan: 0 to add, 1 to change, 0 to destroy.
```

### Important Factoids

The `secgroups.AddServer` is executed only during the update func:
https://github.com/terraform-providers/terraform-provider-openstack/blob/b1406b8e4894faad993aff786f0bb50bfec8e281/openstack/resource_openstack_compute_instance_v2.go#L778

I assume it should be executed during the create func as well.

**Comment by jtopjian** [comment]
@kayrus Quick question: are you sure this isn't a case of mixing security groups, instances, and ports? This behavior is documented here: https://www.terraform.io/docs/providers/openstack/r/compute_instance_v2.html#instances-and-ports

**Comment by kayrus** [comment]
Well, nevertheless it looks like an idempotency problem. I haven't specified security groups for the port, however I specified security groups for the server, and they are applied only on the second terraform run. This is not a serious bug, and having the #457 flying around, you can just remember this use case.

**Comment by jtopjian** [comment]
> Well, nevertheless it looks like an idempotency problem.

I agree.

This topic comes up once a year or so. It's annoying and I certainly sympathize with that. I've spent many hours looking into how this can be resolved and I'm pretty sure (always room for error) that it's not possible. There's the interaction of Neutron and Nova combined with possible variations of this behavior depending on the OpenStack release (due to Nova's security group proxying) that all complicate it.

> This is not a serious bug, and having the #457 flying around, you can just remember this use case.

Agreed. I've made a comment on #457 to be mindful of this issue.

I'm going to close this issue for now, though.
